api/src/main/java/org/openmrs/module/mksreports/reports/PatientHistoryReportManager.java

```diff
@@ -0,0 +1,236 @@
+		encounterAndVistDatasetSetDef.addColumn(VISIT_UUID_LABEL, encounterDataLibrary.getVisitId(), StringUtils.EMPTY,
+		    new VisitUUIDFromIdConverter());
+		encounterAndVistDatasetSetDef.addColumn(VISIT_LOCATION_LABEL, encounterDataLibrary.getVisitId(), StringUtils.EMPTY,
+		    new VisitLocationFromIdConverter());
+		encounterAndVistDatasetSetDef.addColumn(VISIT_TYPE_LABEL, encounterDataLibrary.getVisitId(), StringUtils.EMPTY,
+		    new VisitTypeFromIdConverter());
+		encounterAndVistDatasetSetDef.addColumn(ENCOUNTER_UUID_LABEL, encounterDataLibrary.getUUID(), StringUtils.EMPTY,
+		    new ObjectFormatter());
+		encounterAndVistDatasetSetDef.addColumn(ENCOUNTER_PROVIDER_LABEL, new EncounterIdDataDefinition(), StringUtils.EMPTY,
+		    new EncounterProviderFromIdConverter());
+		encounterAndVistDatasetSetDef.addColumn(ENCOUNTER_TYPE_UUID_LABEL, new EncounterIdDataDefinition(),
+		    StringUtils.EMPTY, new EncounterTypeUUIDFromEncounterIdConverter());
+		encounterAndVistDatasetSetDef.addColumn(ENCOUNTERTYPE_NAME_LABEL, new EncounterTypeDataDefinition(),
+		    StringUtils.EMPTY, new ObjectFormatter());
+		encounterAndVistDatasetSetDef.addColumn(ENCOUNTER_DATETIME_LABEL, new EncounterDatetimeDataDefinition(),
+		    StringUtils.EMPTY, new DateConverter());
+		encounterAndVistDatasetSetDef.addSortCriteria(ENCOUNTER_DATETIME_LABEL, SortCriteria.SortDirection.DESC);
+		return encounterAndVistDatasetSetDef;
+	}
+	
+	/**
+	 * @param i18nTranslator
+	 * @return
+	 */
+	private PatientDataSetDefinition createDemographicsDataSetDefinition(MessageSourceService i18nTranslator) {
+		PatientDataSetDefinition patientDataSetDef = new PatientDataSetDefinition();
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.identifier"),
+		    builtInPatientDataLibrary.getPreferredIdentifierIdentifier());
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.firstname"),
+		    builtInPatientDataLibrary.getPreferredGivenName());
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.lastname"),
+		    builtInPatientDataLibrary.getPreferredFamilyName());
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.dob"),
+		    basePatientDataLibrary.getBirthdate());
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.age"),
+		    basePatientDataLibrary.getAgeAtEndInYears());
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.gender"),
+		    builtInPatientDataLibrary.getGender());
+		addColumn(patientDataSetDef, i18nTranslator.getMessage("mksreports.patienthistory.demographics.fulladdress"),
+		    basePatientDataLibrary.getAddressFull());
+		
+		return patientDataSetDef;
+	}
+	
+	/**
+	 * @return
+	 */
+	private PatientHistoryObsAndEncounterDataSetDefinition createObsAndEncounterDataSetDefinition() {
```

Comment:
Same here, could we use `ObsDataSetDefinition`?

Reply:
also done